fixes/data_submission_recids.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of HEPData.
+# Copyright (C) 2022 CERN.
+#
+# HEPData is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# HEPData is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HEPData; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from datetime import datetime
+import logging
+
+from flask import current_app
+from flask.cli import with_appcontext
+from invenio_db import db
+from sqlalchemy import func
+
+from hepdata.cli import fix
+from hepdata.ext.elasticsearch.api import index_record_ids, push_data_keywords
+from hepdata.modules.records.utils.common import get_record_by_id
+from hepdata.modules.records.utils.doi_minter import generate_dois_for_submission
+from hepdata.modules.records.utils.submission import finalise_datasubmission
+from hepdata.modules.submission.models import HEPSubmission, DataSubmission
+
+logging.basicConfig()
+log = logging.getLogger('data_submission_recids')
+log.setLevel('INFO')
+
+
+@fix.command()
+@with_appcontext
+def fix_data_submission_recids():
+    """Update associated_recids for data submissions with version > 1"""
+    submissions_to_update = HEPSubmission.query \
+        .filter(HEPSubmission.version > 1) \
+        .filter(HEPSubmission.overall_status == 'finished') \
+        .order_by(HEPSubmission.publication_recid, HEPSubmission.version).all()
+
+    log.info(f"Updating data submissions for {len(submissions_to_update)} publications with version > 1")
+
+    for i, hep_submission in enumerate(submissions_to_update):
+        # We have ordered the results so if this is the last entry in
+        # submissions_to_update, or the next entry has a different
+        # publication_recid, then this must be the latest version
+        is_latest = (i + 1 == len(submissions_to_update)) or \
+            submissions_to_update[i+1].publication_recid != hep_submission.publication_recid
```

Comment:
`submissions_to_update` only contains finished submissions (see l51) so if we had v1 (finished), v2 (finished), , v3 (finished), v4 (todo), then only v2 & v3 would appear in the list. v2 would give `is_latest = False` because the next submission has the same `publication_recid`, but v3 would give `is_latest = True` because v4 is not in the list, and so v3 would be indexed. Does that sound right?